channel/jmsjwtexit/README.md

```diff
@@ -0,0 +1,178 @@
+# JmsJwtExit
+
+A sample security exit for authenticating with JWT tokens issued by a third-party issuer. In this demo, the third-party issuer is a KeyCloak server.
+For details on using JWT tokens with MQ, see [Using authentication tokens in an application](https://www.ibm.com/docs/en/ibm-mq/latest?topic=tokens-using-authentication-in-application).
+
+## Overview
+In MQ 9.3.4, a new authentication mechanism has been added into IBM MQ - authenticating with JWT tokens that are issued by a third-party trusted issuer. This sample implements a security exit to provide token authentication capability without making changes to the original messaging application. 
+
+When called as a security exit on a client-connection channel, the sample connects to an trusted token issuer server, retrieves a JWT Token, and adopts a pre-configured user from the token to authenticate to the IBM MQ Queue Manager. The exit will query a token from the token endpoint using CURL, and parse the response to obtain the token to be added into the MQCSP.
+
+## Building
+To build the exit, include the paths to the MQ allclient and jms-api jars. You can use the jars in the MQ /opt/mqm/java/lib directory. Alternatively, get the [IBM MQ allclient.jar](https://mvnrepository.com/artifact/com.ibm.mq/com.ibm.mq.allclient) and the [jms-api.jar](https://mvnrepository.com/artifact/javax.jms/javax.jms-api/2.0.1) from Maven and include the paths to where they are saved in the javac command. 
+
+```
+javac -cp /opt/mqm/java/lib/com.ibm.mq.allclient.jar:./javax.jms-api-2.0.1.jar com/ibm/mq/exits/jms/JmsJwtExit.java
+```
+
+To build the sample application cd to the directory that contains the sample package and include the [IBM MQ allclient.jar](https://mvnrepository.com/artifact/com.ibm.mq/com.ibm.mq.allclient), [jms-api.jar](https://mvnrepository.com/artifact/javax.jms/javax.jms-api/2.0.1) and the [org.json.jar](https://mvnrepository.com/artifact/org.json/json).
+```
+javac -cp ./com.ibm.mq.allclient.jar:./javax.jms-api-2.0.1.jar:./json-20220320.jar com/ibm/mq/samples/jms/JmsPutGet.java
+```
+
+Once built, the exit JmsJwtExit.class file must be copied into the MQ exits64 directory located under the MQ Data Directory (commonly /var/mqm for UNIX).
+
+## Configuration
+
+Configure the queue manager to run the JWT token demo.
+
+On your Linux machine or environment, with IBM MQ installed:
+
+1. Create a queue manager and configure appropriately:
+   1. crtmqm DEMO
+   2. strmqm DEMO
+   3. runmqsc DEMO
+   ```
+
```

Comment:
need to delete this blank line

Reply:
Think this is fixed in one of the later versions.